crates/store/src/db/models/queries/insertions.rs

```diff
@@ -226,6 +281,44 @@ pub(crate) fn upsert_accounts(
 
                 let account = apply_delta(account, delta, &update.final_state_commitment())?;
 
+                // Update assets
+
+                for (faucet_id, _) in delta.vault().fungible().iter() {
+                    let current_amount = account.vault().get_balance(*faucet_id).unwrap();
+                    let asset_update = if current_amount == 0u64 {
+                        None
+                    } else {
+                        Some(Asset::Fungible(
+                            FungibleAsset::new(*faucet_id, current_amount).unwrap(),
+                        ))
+                    };
+                    // TODO: make vault key function public on miden-base
+                    let asset_remove_me = FungibleAsset::new(*faucet_id, 0).unwrap();
+                    insert_account_vault_asset(
+                        conn,
+                        account.id(),
+                        block_num,
+                        asset_remove_me.vault_key(),
+                        asset_update,
+                    )
+                    .unwrap();
```

Comment:
Let's make this change ASAP as I'm hoping to release `miden-base` today.

Reply:
This works for fungible faucets, but not as well for sharing the API with non fungible faucets because we need a new way to specify the removals.